nautobot/extras/models/metadata.py

```diff
@@ -135,3 +141,298 @@ def save(self, *args, **kwargs):
+            ),
+            models.Index(
+                name="assigned_object_contact",
+                fields=["assigned_object_type", "assigned_object_id", "contact"],
+            ),
+            models.Index(
+                name="assigned_object_team",
+                fields=["assigned_object_type", "assigned_object_id", "team"],
+            ),
+        ]
+
+    @property
+    def value(self):
+        if self.metadata_type.data_type == MetadataTypeDataTypeChoices.TYPE_CONTACT_TEAM:
+            return self.contact or self.team
+        else:
+            return self._value
+
+    @value.setter
+    def value(self, v):
+        if self.metadata_type.data_type == MetadataTypeDataTypeChoices.TYPE_CONTACT_TEAM:
+            if isinstance(v, Contact):
+                self.contact = v
+                self.team = None
+            elif isinstance(v, Team):
+                self.team = v
+                self.contact = None
+            else:
+                raise ValidationError(
+                    f"{v} is an invalid value for metadata type data type {self.metadata_type.data_type}"
+                )
+        else:
+            self._value = v
+        self.clean()
+
+    def __str__(self):
+        return f"{self.metadata_type} - {self.assigned_object} - {self.value}"
+
+    def clean(self):
+        """
+        Validate a value according to the field's type validation rules.
+
+        Returns the value, possibly cleaned up
+        """
+        super().clean()
+        value = self._value
+        metadata_type_data_type = self.metadata_type.data_type
+        if (
+            isinstance(value, list)
+            and len(value) == 1
+            and self.metadata_type.data_type != MetadataTypeDataTypeChoices.TYPE_MULTISELECT
+        ):
+            value = value[0]
+        # Validate the assigned_object_type is allowed in metadata_type's content_types
+        if not self.metadata_type.content_types.filter(pk=self.assigned_object_type.id).exists():
+            raise ValidationError(
+                f"Assigned Object Type {self.assigned_object_type} is not allowed by Metadata Type {self.metadata_type}"
+            )
+        # Check for MetadataTypeDataTypeChoices.TYPE_CONTACT_TEAM first
+        if metadata_type_data_type == MetadataTypeDataTypeChoices.TYPE_CONTACT_TEAM:
+            if value not in [None, "", []]:
+                raise ValidationError(f"A value cannot be specified for type {metadata_type_data_type}")
+            if self.contact is None and self.team is None:
+                raise ValidationError("Either a contact or a team must be specified")
+            if self.contact is not None and self.team is not None:
+                raise ValidationError("A contact and a team cannot be both specified at once")
+        elif value in [None, "", []]:
+            raise ValidationError(
+                f"value is a required field that cannot be empty for metadata type {metadata_type_data_type}."
+            )
```

Comment:
Do we want to allow TYPE_JSON at least to be an empty string or empty list? What about an empty string for TYPE_TEXT or an empty list for type TYPE_MULTISELECT?

Reply:
changed this to `elif value is None:` to allow "" and [].